File: min_d_and_c.cpp
#include<bits/stdc++.h>
using namespace std;

int arr[12]={234,43,523,4,23,324,3332,4,234,3,4,24};


pair<int,int> min_and_max(int l,int r){

    if (l==r) return {arr[l],arr[l]};
    int mid = (l+r)/2;
    pair<int,int> left = min_and_max(l,mid);
    pair<int,int> right = min_and_max(mid+1,r);
    return {min(left.first,right.first),max(left.second,right.second)};
}

int max(int l,int r){

    if (l==r) return arr[l];
    int mid = (l+r)/2;
    return max(max(l,mid),max(mid+1,r));
}

int min(int l,int r){

    if (l==r) return arr[l];
    int mid = (l+r)/2;
    return min(min(l,mid),min(mid+1,r));
}

int main(){


int n= sizeof(arr)/sizeof(arr[0]);
pair<int,int> ans = min_and_max(0,n-1);
cout<<"Minimum element is: "<<ans.first<<endl;
cout<<"maximum element is: "<<ans.second<<endl;


}

File: min_d_and_c.py
arr=[432,54,21,4,34,4,243,45,34423,423,4234,3,243,35]
l=0
r=len(arr)-1
def dc(l,r):
    if l==r:
        return arr[l]
    mid=(l+r)//2
    left=dc(l,mid)
    right=dc(mid+1,r)
    return min(left,right)
print(dc(l,r))

def max_dc(l,r):
    if l==r:
        return arr[l]
    mid=(l+r)//2
    left=max_dc(l,mid)
    right=max_dc(mid+1,r)
    return max(left,right)
print(max_dc(l,r))


def min_max(l,r):
    if l==r:
        return arr[l],arr[l]
    mid=(l+r)//2
    left_min,left_max=min_max(l,mid)
    right_min,right_max=min_max(mid+1,r)
    return min(left_min,right_min),max(left_max,right_max)
print(min_max(l,r))
